### PR TITLE
feat: restrict event for ongoing business process on case (CMC-799)

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[CMC-799](url)

### Change description ###
This PR restricts event if there is ongoing business process on the case. 

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcess.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.unspec.aspect;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface NoOnGoingBusinessProcess {
```
Comment (swalker125) [inline src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcess.java]: Ongoing is one word :)

File: src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcessAspect.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                "%s is not allowed on the case %s due to ongoing business process",
+                caseEvent.getDisplayName(), caseData.getCcdCaseReference()
+            ));
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(List.of(ERROR_MESSAGE))
+                .build();
+        }
+    }
+
+    private boolean hasNoOnGoingBusinessProcess(CaseData caseData) {
```
Comment (swalker125) [inline src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcessAspect.java]: ongoing one word

**Review by swalker125 — COMMENTED** [review]
LGTM, only change would be OnGoing -> Ongoing everywhere

File: src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcessAspect.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import static java.lang.String.format;
+import static uk.gov.hmcts.reform.unspec.enums.BusinessProcessStatus.FINISHED;
+
+@Slf4j
+@Aspect
+@Component
+@RequiredArgsConstructor
+public class NoOnGoingBusinessProcessAspect {
+
+    public static final String ERROR_MESSAGE = "There is a technical issue causing a delay. "
```
Comment (pd-dudkiewicz) [inline src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcessAspect.java]: Should we include that the error is caused by ongoing business process in message to the user?

**Review by pd-dudkiewicz — DISMISSED** [review]
looks good to me 👍 just one minor comment

File: src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcessAspect.java (hunk 3 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcessAspect.java]: This message is provided by the content team in this contect ticket: CMC-800

File: src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcessAspect.java (hunk 2 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOnGoingBusinessProcessAspect.java]: renamed

**Review by pd-dudkiewicz — APPROVED** [review]

**Review by douglasrice — APPROVED** [review]

**Pull request merged**